code/datums/components/riding.dm

```diff
@@ -1,65 +1,116 @@
+/**
+  * This is the riding component, which is applied to a movable atom by the [ridable element][/datum/element/ridable] when a mob is successfully buckled to said movable.
```

Comment:
I don't see any big benefits to this, if anything it'll complicate things more than they already are with riding. Is the reasoning memory overhead or?

Reply:
I never got around to answering this, but being able to separate the code for "can I actually ride this" and "I am actually riding this" was a very big help in figuring out what checks belong on the atoms themselves whether it's even possible to buckle the rider to the ridden (are there seats available, are we actually dead and not capable of buckling ourselves, etc), what checks should take place on the element once we've determined that we're past the standard buckled hurdles and are halfway capable of riding this (do we have the hands available for the kind of riding we want), and the code handling the actual riding instance once we've mounted it.

I don't at all think separating it into an element and a component complicates things. I've found it very helpful to be able to count on elements representing the final checks for whether something is mountable while components represent the actual act of riding. Perhaps it's possible to distill them further, but I'm very happy with these steps.